### PR TITLE
Activate and use bulk upsert

### PR DESCRIPTION
**Pull request by lippserd**
closes #298

### CODE REVIEW

File: pkg/icingadb/db.go [hunk 1]
```diff
@@ -98,7 +98,7 @@ func (db *DB) BuildUpdateStmt(update interface{}) string {
 		`UPDATE %s SET %s WHERE id = :id`,
 		utils.TableName(update),
 		strings.Join(set, ", "),
-	)
+	), len(columns)
```
Comment (julianbrost) [inline pkg/icingadb/db.go:101]: With `:id` in the `WHERE` clause, there is one more placeholder.

File: pkg/icingadb/db.go [hunk 2]
```diff
@@ -371,7 +371,9 @@ func (db *DB) CreateStreamed(ctx context.Context, entities <-chan contracts.Enti
 	}
 
 	sem := db.getSemaphoreForTable(utils.TableName(first))
-	return db.NamedBulkExec(ctx, db.BuildInsertStmt(first), 1<<15/len(db.BuildColumns(first)), sem, forward, nil)
+	stmt, placeholders := db.BuildInsertStmt(first)
+
+	return db.NamedBulkExec(ctx, stmt, 1<<15/placeholders, sem, forward, nil)
```
Comment (julianbrost) [inline pkg/icingadb/db.go:376]: `1<<15` could be made a named constant to better explain why this is done here (not directly related to this PR, but if you're touching the code anyways).

**Review by julianbrost — CHANGES_REQUESTED** [review]

File: pkg/icingadb/db.go (hunk 1 above)
Comment (lippserd) [reply pkg/icingadb/db.go]: True!

File: pkg/icingadb/db.go (hunk 2 above)
Comment (lippserd) [reply pkg/icingadb/db.go:376]: Will be addressed in the code-quality PR.

**Review by julianbrost — APPROVED** [review]

**Pull request merged**